go-controller/pkg/ovn/pods.go

```diff
@@ -261,6 +263,17 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 	annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
 
+	defer func(nodeName string) {
```

Comment:
can you add some comments explaining the logic?

Reply:
done.